src/guide/receive_payment.md

```diff
@@ -1,8 +1,177 @@
+1. [Receiving the Payment](receive_payment.md#receiving-payments-1)
+
+## Preparing Payments
+During the prepare step, the SDK ensures that the inputs are valid with respect to the specified payment method,
+and also returns the relative fees related to the payment so they can be confirmed.
+
+
+The SDK currently supports three methods of receiving: Lightning, Liquid and Cross-chain (Bitcoin):
+
+### Lightning
+When receiving via Lightning, we generate an invoice to be paid.  Note that the payment may fallback to a direct Liquid payment (if the payer's client supports this).
+
+
+**Note:** The amount field is currently mandatory when paying via Lightning.
+
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust,ignore
+{{#include ../../snippets/rust/src/receive_payment.rs:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">Swift</div>
+<section>
+
+```swift,ignore
+{{#include ../../snippets/swift/BreezSDKExamples/Sources/ReceivePayment.swift:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">Kotlin</div>
+<section>
+
+```kotlin,ignore
+{{#include ../../snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ReceivePayment.kt:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">React Native</div>
+<section>
+
+```typescript
+{{#include ../../snippets/react-native/receive_payment.ts:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">Dart</div>
+<section>
+
+```dart,ignore
+{{#include ../../snippets/dart_snippets/lib/receive_payment.dart:prepare-receive-payment-lightning}}
+```
+</section>
+</custom-tabs>
+
+### Liquid
+When receiving via Liquid, we can either generate an address to receive to, or a BIP21 URI with information regarding the payment (currently only the amount and message).
+
+
+To generate a BIP21 address, all you have to do is specify a payer amount.
+
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust,ignore
+{{#include ../../snippets/rust/src/receive_payment.rs:prepare-receive-payment-liquid}}
+```
+</section>
+
+<div slot="title">Swift</div>
+<section>
+
+```swift,ignore
+{{#include ../../snippets/swift/BreezSDKExamples/Sources/ReceivePayment.swift:prepare-receive-payment-liquid}}
+```
+</section>
+
+<div slot="title">Kotlin</div>
+<section>
+
+```kotlin,ignore
+{{#include ../../snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ReceivePayment.kt:prepare-receive-payment-liquid}}
+```
+</section>
+
+<div slot="title">React Native</div>
+<section>
+
+```typescript
+{{#include ../../snippets/react-native/receive_payment.ts:prepare-receive-payment-liquid}}
+```
+</section>
+
+<div slot="title">Dart</div>
+<section>
+
+```dart,ignore
+{{#include ../../snippets/dart_snippets/lib/receive_payment.dart:prepare-receive-payment-liquid}}
+```
+</section>
+</custom-tabs>
+
+### Cross-chain (Bitcoin)
```

Comment:
Can do move it above the Liquid section? 

I think its ok to remove the "Cross-chain", I think it's enough if we describe the functioning of the SDK elsewhere where it uses submarine/chain swaps.
```suggestion
### Bitcoin
```